File: ove/flutter/lib/domain/model/session.dart
import 'package:ove/domain/model/restatement.dart';
import 'package:ove/domain/model/belief_selection.dart';
import 'package:ove/domain/model/action_item.dart';

enum SessionStatus { inProgress, completed, interrupted }

class LocalSession {
  final String id;
  final DateTime startedAt;
  final DateTime? completedAt;
  final SessionStatus status;
  final String transcript;
  final Restatement? restatement;
  final List<FollowUpQA> followUpQA;
  final BeliefSelection? beliefSelection;
  final List<ActionItem> actionItems;

  const LocalSession({
    required this.id,
    required this.startedAt,
    this.completedAt,
    required this.status,
    this.transcript = '',
    this.restatement,
    this.followUpQA = const [],
    this.beliefSelection,
    this.actionItems = const [],
  });

  LocalSession copyWith({
    String? transcript,
    Restatement? restatement,
    List<FollowUpQA>? followUpQA,
    BeliefSelection? beliefSelection,
    List<ActionItem>? actionItems,
    SessionStatus? status,
    DateTime? completedAt,
  }) =>
      LocalSession(
        id: id,
        startedAt: startedAt,
        completedAt: completedAt ?? this.completedAt,
        status: status ?? this.status,
        transcript: transcript ?? this.transcript,
        restatement: restatement ?? this.restatement,
        followUpQA: followUpQA ?? this.followUpQA,
        beliefSelection: beliefSelection ?? this.beliefSelection,
        actionItems: actionItems ?? this.actionItems,
      );
}

class FollowUpQA {
  final String targetElement; // 'situation' | 'thought' | 'emotion'
  final String question;
  final String answerTranscript;

  const FollowUpQA({
    required this.targetElement,
    required this.question,
    required this.answerTranscript,
  });
}

File: ove/web/lib/models/session.ts
import type { Restatement } from './restatement';
import type { ActionItem } from './action-item';
import type { BeliefSelection } from './belief-selection';
import type { Homework } from './homework';

export type SessionStatus = 'inProgress' | 'completed' | 'interrupted';

export interface FollowUpQA {
  targetElement?: 'situation' | 'thought' | 'emotion';
  question: string;
  answerText: string;
  phase: 'ste' | 'socratic';
  socraticRound?: 1 | 2;
}

export interface LocalSession {
  id: string;
  startedAt: string;
  completedAt?: string;
  status: SessionStatus;
  transcript: string;
  restatement?: Restatement;
  followUpQA: FollowUpQA[];
  empathyResponse?: string;
  beliefHypothesis?: { text: string; belief: string };
  beliefSelection?: BeliefSelection;
  actionItems: ActionItem[];
  homework?: Homework;
}
